### PR TITLE
lr scheduler can now monitor other fields

### PR DESCRIPTION
**Pull request by benjamin-work**
`LRScheduler` now allows you to monitor vars other than train_loss.

The class now takes a 'monitor' argument (like `Checkpoint`). If this is
a string, this value from the history is passed to step. If it is a
callable, the value returned by calling this callable with net is is
passed to step.

Moreover, some minor clean ups.

### CODE REVIEW

File: skorch/callbacks/lr_scheduler.py [hunk 1]
```diff
@@ -34,13 +28,19 @@ class LRScheduler(Callback):
     policy : str or _LRScheduler class (default='WarmRestartLR')
       Learning rate policy name or scheduler to be used.
 
+    monitor : str or callable (default=None)
+      Value of the history to monitor or callback that determines whether
+      this epoch should to a checkpoint. The callback takes the network
+      instance as parameter.
```
Comment (ottonemo) [inline skorch/callbacks/lr_scheduler.py]: Maybe add a note what values to expect (in case of `Checkpoint` we want boolean values, here we want real values, right?)

**Comment by benjamin-work** [comment]
@ottonemo pls review again.

File: skorch/callbacks/lr_scheduler.py [hunk 2]
```diff
@@ -29,9 +29,10 @@ class LRScheduler(Callback):
       Learning rate policy name or scheduler to be used.
 
     monitor : str or callable (default=None)
-      Value of the history to monitor or callback that determines whether
-      this epoch should to a checkpoint. The callback takes the network
-      instance as parameter.
+      Value of the history to monitor or function/callable that. In
```
Comment (ottonemo) [inline skorch/callbacks/lr_scheduler.py]: -that

Comment (benjamin-work) [reply skorch/callbacks/lr_scheduler.py]: Can't write...
